src/test/java/com/ihren/task2/exercise6/MetricServiceTest.java

```diff
@@ -0,0 +1,228 @@
+package com.ihren.task2.exercise6;
+
+import com.ihren.task2.exercise6.model.Counter;
+import com.ihren.task2.exercise6.model.Metrics;
+import com.ihren.task2.exercise6.model.Timer;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockedStatic;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+import static org.mockito.BDDMockito.willThrow;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.never;
+
+@ExtendWith(MockitoExtension.class)
+class MetricServiceTest {
```

Comment:
You have a lot of extra test cases. You should have four tests for each methods.